src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -706,10 +754,49 @@ public void saveResolutions()
           }
         }
       }
+
       // Wait until the entire resolution has completed before proceeding to
       // the next one
       CompletableFuture.allOf(
         futures.toArray(new CompletableFuture[futures.size()])).join();
+
+      // TODO: some of these futures may be completelyExceptionally
+      //  and need re-throwing
+
+    }
+
+  }
+
+  /**
+   * Takes exception from asynchronous execution and re-throw known exception
+   * types. If the end is reached with no known exception detected, either the
+   * exception itself will be thrown if {@link RuntimeException}, otherwise
+   * wrap in a {@link RuntimeException}.
+   *
+   * @param t Exception raised during processing.
+   */
+  private void unwrapException(Throwable t)
```

Comment:
Is this method used anywhere?

Reply:
How did _that_ happen?! Must have gotten lost in the cleanup. I'll push it with the rest of the fixes.